src/app/pages/salidas/form-elementos-seleccionados/form-elementos-seleccionados.component.ts

```diff
@@ -40,149 +19,176 @@ import { isObject } from 'rxjs/internal-compatibility';
     }
+  }
 
+  private cambiosFuncionario(control: AbstractControl): Observable<Partial<TerceroCriterioContratista>[]> {
+    return control.valueChanges
+      .pipe(
+        startWith(''),
+        map(val => typeof val === 'string' ? val : this.muestraFuncionario(val)),
+        map(nombre => this.filtroFuncionarios(nombre)),
+      );
   }
 
-  // MÉTODO PARA BUSCAR PROVEEDORES EXISTENTES
-  loadProveedoresElementos(): void {
-    if (this.proveedorfiltro.length > 3) {
-      this.terceros.getProveedores(this.proveedorfiltro).subscribe(res => {
-        if (res != null) {
-          while (this.elementos.length > 0) {
-            this.elementos.pop();
-          }
-          if (Object.keys(res).length === 1) {
-            this.elementos2 = res[0];
-            // console.log(res[0]);
-          }
-          for (const index of Object.keys(res)) {
-            if (res[index].NombreCompleto != null) {
-              this.elementos.push(res[index].NombreCompleto);
-            }
-          }
-        }
-      });
+  private cambiosDependencia(valueChanges: Observable<any>) {
+    valueChanges.pipe(
+      startWith(''),
+      debounceTime(250),
+      distinctUntilChanged(),
+      switchMap((val) => this.loadDependencias(val) ),
+    ).subscribe((response: any) => {
+      this.dependencias = response.queryOptions[0].Id ? response.queryOptions : [];
+      this.getUbicaciones();
+    });
+  }
 
-    }
+  private filtroFuncionarios(nombre: string): TerceroCriterioContratista[] {
+    if (nombre.length >= 4 && Array.isArray(this.Funcionarios)) {
+      const valorFiltrado = nombre.toLowerCase();
+      return this.Funcionarios.filter(contr => this.muestraFuncionario(contr).toLowerCase().includes(valorFiltrado));
+    } else return [];
+  }
+
+  private loadDependencias(text: string) {
+    const queryOptions$ = text.length > 3 ?
+      this.oikosHelper.getDependencias(text) :
+      new Observable((obs) => { obs.next([{}]); });
+    return combineLatest([queryOptions$]).pipe(
+      map(([queryOptions_$]) => ({
+        queryOptions: queryOptions_$,
+      })),
+    );
   }
 
-  // MÉTODO PARA FILTRAR PROVEEDORES
-  changeNombreProveedor(event) {
-    this.proveedorfiltro = event.target.value;
-    this.loadProveedoresElementos();
+  private loadFuncionarios(): Promise<void> {
+    return new Promise<void>(resolve => {
+      this.tercerosHelper.getTercerosByCriterio('funcionarios').toPromise().then(res => {
+        this.Funcionarios = res;
+        resolve();
+      });
+    });
+  }
 
+  private validarTercero(): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+      const valor = control.value;
+      const checkStringLength = typeof (valor) === 'string' && valor.length < 4 && valor !== '' ? true : false;
+      const checkInvalidString = typeof (valor) === 'string' && valor !== '' ? true : false;
+      const checkInvalidTercero = typeof (valor) === 'object' && !valor.Tercero ? true : false;
```

Comment:
Uso innecesario de ternas, de hecho una terna funciona con un booleano por lo que lo que está a la izquierda ya sería un booleano
```suggestion
      const checkStringLength = typeof (valor) === 'string' && valor.length < 4 && valor !== '';
      const checkInvalidString = typeof (valor) === 'string' && valor !== '';
      const checkInvalidTercero = typeof (valor) === 'object' && !valor.Tercero;
```
Nota: Cuando se requiera negar, agrupar todo entre paréntesis y colocar el operador `!` al comienzo